Toms-Storage-121/src/platform-shared/java/com/tom/storagemod/screen/AbstractStorageTerminalScreen.java

```diff
@@ -76,6 +74,9 @@ public List<String> load(StoredItemStack key) throws Exception {
 		}
 
 	});
+	private static final LoadingCache<StoredItemStack, String> componentCache = CacheBuilder.newBuilder().expireAfterAccess(5, TimeUnit.SECONDS).build(CacheLoader.from(
+			key -> DataComponentPatch.CODEC.encodeStart(JsonOps.COMPRESSED, key.getStack().getComponentsPatch()).mapOrElse(JsonElement::toString, e -> "")
```

Comment:
This will fail with enchanted items.
```java
var ctx = Minecraft.getInstance().level.registryAccess().createSerializationContext(JsonOps.COMPRESSED);
return DataComponentPatch.CODEC.encodeStart(ctx, key.getStack().getComponentsPatch()).mapOrElse(JsonElement::toString, e -> "");
```
I'll add this after merging.

Reply:
I'm actually surprised by this... Aren't enchantment component also a part of the stack component patch?
I mean according to [the wiki](https://minecraft.wiki/w/Enchanting#Item_data) it is a part of the stack component, and in order to have a stack-specific value, need to be a part of the patch, right?

Reply:
Actually looking at [your changes](https://github.com/tom5454/Toms-Storage/blob/575304f54555a8f55e40a171b38bf03ad7f12361/Toms-Storage-121/src/platform-shared/java/com/tom/storagemod/screen/AbstractStorageTerminalScreen.java#L85) I think I get it, no worries.
But then did you also apply those changes for the 1.20.6 version?

Reply:
Yes I applied it before pushing the update.
Enchanted items need access to the enchantments registry for it to serialize correctly.

Reply:
Oh, right, with the 1.21 enchantment overhaul that makes sense indeed